src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Fabric;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.ObjectPool;
+using ServiceFabricHealth = System.Fabric.Health;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal abstract class HealthCheckPublisher : IHealthCheckPublisher
```

Comment:
Make sure you add unit tests

Reply:
HealthCheckPublisher is covered by currently existing tests

Reply:
Maybe we can add tests for RestPublisher - in that case I think that we should reuse courrent tests, maybe inherit them? Not sure, probably worth checking.